### PR TITLE
Restructured bottles of beer recursive example

### PR DESCRIPTION
**Pull request by drewesk**
- Changed structure of song to better match the pauses of singing the song
- Made my base_case def last in order to avoid hoisting
- Var names differ

[#131780287]

### CODE REVIEW

File: methods_and_recursion/bottles_of_beer.rb [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    "#{bottles} #{words}"
+  end
+end
+
+def no_bottles_left
+  puts 'No more bottles of beer on the wall, no more bottles of beer.'
+  puts 'Go to the store and buy some more, 99 bottles of beer on the wall.'
+end
+
+def base_case(bottles)
```
Comment (jasonnoble) [inline methods_and_recursion/bottles_of_beer.rb]: Wouldn't a better name be `sing_verse`?

File: methods_and_recursion/bottles_of_beer.rb [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  bottle_of_beer = "#{plural(bottles, 'bottle')} of beer"
+  puts bottle_of_beer + ' on the wall, ' + bottle_of_beer + '.'
```
Comment (jasonnoble) [inline methods_and_recursion/bottles_of_beer.rb:15]: Does `puts "#{plural(bottles, 'bottle') of beer"` not work?

Comment (jasonnoble) [reply methods_and_recursion/bottles_of_beer.rb:15]: Oh, I see what you did...  Carry on. :)

**Review by jasonnoble — CHANGES_REQUESTED** [review]

File: methods_and_recursion/bottles_of_beer.rb (hunk 2 above)
Comment (drewesk) [reply methods_and_recursion/bottles_of_beer.rb:15]: Damn rubocop putting on the shackles!

File: methods_and_recursion/bottles_of_beer.rb [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+  words += 's' if bottles != 1
+  if bottles.zero?
+    'No more bottles'
+  else
+    "#{bottles} #{words}"
+  end
+end
+
+def no_bottles_left
+  puts 'No more bottles of beer on the wall, no more bottles of beer.'
```
Comment (jasonnoble) [inline methods_and_recursion/bottles_of_beer.rb:9]: So, on line 23, you check to see if bottles > 1, if it's not, i.e. it's the last bottle, you call `no_bottles_left`.  Which means you don't need the if check on line 5...

**Review by jasonnoble — CHANGES_REQUESTED** [review]

**Review by jasonnoble — APPROVED** [review]

**Pull request merged**